src/mxcp/server/interfaces/server/mcp.py

```diff
@@ -720,51 +721,54 @@ async def shutdown(self) -> None:
-            # This handles python runtimes, plugins, and the db session.
+        # Close OAuth server persistence
+        if self.oauth_server:
             try:
-                self.shutdown_runtime_components()
+                await asyncio.wait_for(self.oauth_server.close(), timeout=5.0)
+                logger.info("Closed OAuth server")
+            except asyncio.TimeoutError:
+                logger.error("OAuth server shutdown timed out after 5 seconds")
             except Exception as exc:
-                logger.error(f"Error shutting down runtime components: {exc}")
-
-            # Close OAuth server persistence
-            if self.oauth_server:
-                try:
-                    await asyncio.wait_for(self.oauth_server.close(), timeout=5.0)
-                    logger.info("Closed OAuth server")
-                except asyncio.TimeoutError:
-                    logger.error("OAuth server shutdown timed out after 5 seconds")
-                except Exception as exc:
-                    logger.error(f"Error closing OAuth server: {exc}")
+                logger.error(f"Error closing OAuth server: {exc}")
 
-            # Shutdown audit logger if initialized
-            if self.audit_logger:
-                try:
-                    await asyncio.wait_for(self.audit_logger.close(), timeout=5.0)
-                    logger.info("Closed audit logger")
-                except asyncio.TimeoutError:
-                    logger.error("Audit logger shutdown timed out after 5 seconds")
-                except Exception as exc:
-                    logger.error(f"Error closing audit logger: {exc}")
-
-            # Shutdown telemetry (synchronous)
+        # Shutdown audit logger if initialized
+        if self.audit_logger:
             try:
-                shutdown_telemetry()
-                logger.info("Shutdown telemetry")
+                await asyncio.wait_for(self.audit_logger.close(), timeout=5.0)
+                logger.info("Closed audit logger")
+            except asyncio.TimeoutError:
+                logger.error("Audit logger shutdown timed out after 5 seconds")
             except Exception as exc:
-                logger.error(f"Error shutting down telemetry: {exc}")
+                logger.error(f"Error closing audit logger: {exc}")
 
+        # Shutdown telemetry (synchronous)
+        try:
+            shutdown_telemetry()
+            logger.info("Shutdown telemetry")
         except Exception as exc:
-            logger.error(f"Error during shutdown: {exc}")
-        finally:
-            logger.info("MXCP server shutdown complete")
+            logger.error(f"Error shutting down telemetry: {exc}")
+
+        # Flush analytics events (synchronous)
+        try:
+            flush_analytics()
+            logger.info("Flushed analytics")
+        except Exception as exc:
+            logger.error(f"Error flushing analytics: {exc}")
```

Comment:
Elsewhere we hide errors coming from analytics, we don't log (in analytics/init: `# Silently fail - analytics should never affect the main application`).

We probably should also hide that error too?